extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -318,38 +318,16 @@ public static int parseDurationString(@Nonnull final String input)
                 ? input.split(":")
                 : input.split("\\.");
 
-        String days = "0";
-        String hours = "0";
-        String minutes = "0";
-        final String seconds;
-
-        switch (splitInput.length) {
-            case 4:
-                days = splitInput[0];
-                hours = splitInput[1];
-                minutes = splitInput[2];
-                seconds = splitInput[3];
-                break;
-            case 3:
-                hours = splitInput[0];
-                minutes = splitInput[1];
-                seconds = splitInput[2];
-                break;
-            case 2:
-                minutes = splitInput[0];
-                seconds = splitInput[1];
-                break;
-            case 1:
-                seconds = splitInput[0];
-                break;
-            default:
-                throw new ParsingException("Error duration string with unknown format: " + input);
-        }
-
-        return ((convertDurationToInt(days) * 24
-                + convertDurationToInt(hours)) * 60
-                + convertDurationToInt(minutes)) * 60
-                + convertDurationToInt(seconds);
+        final int[] units = {24, 60, 60, 1};
+        final int offset = units.length - splitInput.length;
+        if (offset < 0) {
```

Comment:
In the case statement being replaced, the valid values were 1, 2, 3, and 4. Consider changing this if statement to:

if (offset < 1 || offset > 4)

This should preserve the same logic that existed in the if statement.

Reply:
> In the case statement being replaced, the valid values were 1, 2, 3, and 4. Consider changing this if statement to:
> 
> if (offset < 1 || offset > 4)
> 
> This should preserve the same logic that existed in the if statement.

@acrodemocide `splitInput.length` is a number strictly greater than 0, while `units.length` is equal to (the magic number) 4.
Then `offset` cannot be a number strictly greater than 4.

Reply:
Thank you for clarifying.